server/session.go

```diff
@@ -320,22 +328,66 @@ func (p *Plugin) removeSession(us *session) error {
 	if prevState.Call != nil && prevState.Call.Recording != nil && currState.Call != nil && currState.Call.Recording != nil &&
 		currState.Call.Recording.EndAt > prevState.Call.Recording.EndAt {
 
-		p.LogDebug("recording bot left the call, attempting to stop job", "channelID", us.channelID, "jobID", currState.Call.Recording.JobID)
+		p.LogDebug("recording bot left the call", "channelID", us.channelID, "jobID", currState.Call.Recording.JobID)
 
 		// Since MM-52346 we don't need to explicitly stop the recording here as
 		// the bot leaving the call will implicitly terminate the recording process.
 
+		if currState.Call.Transcription != nil && currState.Call.Transcription.EndAt == 0 {
+			p.LogDebug("attempting to stop transcribing job", "channelID", us.channelID, "jobID", currState.Call.Transcription.JobID)
+			if err := p.stopTranscribingJob(currState, us.channelID); err != nil {
+				p.LogError("failed to stop transcription", "channelID", us.channelID, "err", err.Error())
+			}
+		}
+
 		p.publishWebSocketEvent(wsEventCallRecordingState, map[string]interface{}{
 			"callID":   us.channelID,
 			"recState": currState.Call.Recording.getClientState().toMap(),
 		}, &model.WebsocketBroadcast{ChannelId: us.channelID, ReliableClusterSend: true})
 	}
 
-	// If the bot is the only user left in the call we automatically stop the recording.
-	if currState.Call != nil && currState.Call.Recording != nil && currState.Call.onlyUserLeft(p.getBotID()) {
-		p.LogDebug("all users left call with recording in progress, stopping", "channelID", us.channelID, "jobID", currState.Call.Recording.JobID)
-		if err := p.getJobService().StopJob(us.channelID); err != nil {
-			p.LogError("failed to stop recording job", "error", err.Error(), "channelID", us.channelID, "jobID", currState.Call.Recording.JobID)
+	// Checking if the transcription has ended due to the bot leaving.
+	if prevState.Call != nil && prevState.Call.Transcription != nil && currState.Call != nil && currState.Call.Transcription != nil &&
+		currState.Call.Transcription.EndAt > prevState.Call.Transcription.EndAt {
```

Comment:
I'm starting to think the way that the bot behaviour + state is triggering things seems a little brittle and confusing.  Maybe it's too late to change it, but it would be nice if jobs were started and stopped by idempotent events. Granted, we need the bots to do any of the actual recording/transcribing, but it seems like maybe the communication triggers and the stopping triggers (at least) could be in response to events, rather than these  conditional blocks (which seems like it would be error prone).

Reply:
I'd like to understand more about that as I am not entirely sure what you mean by "idempotent events". Could you make an example perhaps?

In essence the code above is in response to an event, namely the bot leaving the call which we need to handle somehow. Not sure if it's a syntactic improvement you'd be looking for (e.g. `onBotLeftCall` callback) or something more structural.

Reply:
Here's my thinking (and while writing this I think I see why it's not practical): Something like moving all the "bot leaving triggers the recording/transcription ending" logic out of here and putting it in its own function (say: `recordingEnded`), then moving the conditional logic (lines 350-351) somewhere else (that's the infeasible part I think), then using that conditional logic to trigger an api call which trigger the `recordingEnded`. Then if that `recordingEnded` endpoint gets triggered multiple times (for whatever reason) we can ignore it after the first one. 
I think it's tough to find a better place for lines 350-351 though...
I guess I'm worried that we've tightly coupled the bot's presence with the recording/transcribing activity. I know we need the bot there for that to actually work, I just feel it's mixing two concepts that logically should be separate...

Reply:
Alright, I feel this deserves a sync discussion, let's chat about it next week.